### DISCUSSION TITLE
Change BC only at a particular time

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18027] [Q&A General] [answered]
Hi guys,

I want to change the boundary condition of the sides of the subdomain ONLY at time X (before time X let it equal to whatever is calculated from the simulation)

```
[./water_grad_block7]
type = FunctionDirichletBC
variable = porepressure
boundary = 'front7 top7 right7 back7 bottom7 left7'
function = pp_block7
[../]

[./pp_block7]
  type = ParsedFunction
  value = 'if(t<4,xxxxx,0)'
  [../]
```

xxxx = calculated pore-pressure from simulation

Is this possible with MOOSE?

Kind regards,
Traiwit

**Comment by GiudGiud** [comment]
This should be possible using a postprocessor for xxxx and having it be an argument (in vals) of the ParsedFunction

https://mooseframework.inl.gov/source/functions/MooseParsedFunction.html

**Reply by Traiwit** [reply]
Hi @GiudGiud,

I was looking at this 'postprocessor' as well. However, from what I see, I couldn't find the right one that can store the variable of each side of the sub-domain.

@WilkAndy have you ever encountered this situation?

Kind regards,
Traiwit

**Comment by WilkAndy** [comment, marked as answer]
Have a look half-way down this page, for BCs and Controls:

https://mooseframework.inl.gov/modules/porous_flow/ates.html

a

**Reply by Traiwit** [reply]
Hi @WilkAndy and @GiudGiud 

I got it to work now with the control block
![image](https://user-images.githubusercontent.com/79562439/121302369-ed37a600-c93c-11eb-9980-3c4f3c5ccb15.png)


Thank you guys.

Kind regards,
Traiwit
